packages/core/src/dialects/postgres/connection-manager.ts

```diff
@@ -57,7 +58,7 @@ export class PostgresConnectionManager extends AbstractConnectionManager<PgConne
   #oidMap = new Map<number, TypeOids>();
   #oidParserCache = new Map<number, TypeParser<any, any>>();
 
-  constructor(dialect: PostgresDialect, sequelize: Sequelize) {
+  constructor(dialect: PostgresDialect | CockroachDbDialect, sequelize: Sequelize) {
```

Comment:
the postgres connection manager cannot have a direct reference to cockroachdb, it will create an package loop between `@sequelize/postgres` and `@sequelize/cockroachdb`.

After the merge of `main` in this PR, we can look into how this can be solved. I think the solution would be to have `CockroachDbDialect` extend `PostgresDialect`, but that may require moving some init code somehow